app/models/derivative.rb

```diff
@@ -55,12 +53,11 @@ class Derivative < ActiveFedora::Base
     end
   end
 
-  def self.url_handler
-    url_handler_class = Avalon::Configuration.lookup('streaming.server').to_s.classify
-    @url_handler ||= UrlHandler.const_get(url_handler_class.to_sym)
-  end
+  def self.from_output(dists, opts={})
+    #output is an array of 1 or more distributions of the same derivative (e.g. file and HLS segmented file)
+    hls_output = dists.delete(dists.find {|o| o[:url].ends_with? "m3u8" })
+    output = dists.first || hls_output
```

Comment:
So, output will only be set to hls_output if no others exist. This seems like it might not be correct, based on the comment above.